Sources/Utility/ArgumentParser.swift

```diff
@@ -29,6 +29,19 @@ public enum ArgumentParserError: Swift.Error {
     case typeMismatch(String)
 }
 
+/// Various shell completions modes supplied by ArgumentKind.
+///
+/// - none:        offers no completions at all; e.g. for string identifier
+/// - unspecified: no specific completions, will offer tool's completions
+/// - filename:    offers filename completions
+/// - values:      offers completions from predefined list
+public enum ShellCompletion {
+    case none
+    case unspecified
+    case filename
+    case values([(value: String, description: String)])
```

Comment:
Yeah those could definitely be improved. Take for example `swift package init` which can take a `--type` argument. The possible values for this argument along with a description would be written like this:

    let completion = ShellCompletion.values([
        ("empty", "generates an empty project"),
        ("executable", "generates a project for a cli executable"),
        ("library", "generates a project for a dynamic library"),
        ("system-module", "generates a project for a system library"),
    ])